packages/chat-component-bindings/src/handlers/createHandlers.ts

```diff
@@ -28,15 +37,18 @@ export type ChatHandlers = {
     messageId: string,
     content: string,
     /* @conditional-compile-remove(attachment-upload) */
-    options?: {
-      /* @conditional-compile-remove(attachment-upload) */
-      metadata?: Record<string, string>;
-      attachmentMetadata?: AttachmentMetadata[];
-    }
+    options?: UIComponentMessageOptions
   ) => Promise<void>;
   onDeleteMessage: (messageId: string) => Promise<void>;
 };
 
+/**
+ * @beta
+ */
+export type UIComponentMessageOptions = {
+  attachmentMetadata?: AttachmentMetadata[];
```

Comment:
Don't we also need to add `metadata? : Record<string, string>` cause now the signature on the `onUpdateMessage` is different.

Reply:
no, the metadata shouldn't even present in UI library, it's something from SDK where we need to convert attachmentmetadata to. The intention is to have:
- attachmentMetadata as the view model and used everywhere in UI library
- metadata as the model from Chat SDK where it's only used in adapter when calling chat sdk functions

they should not be used interchangeably